lib/core/src/irods_path_recursion.cpp

```diff
@@ -226,7 +226,8 @@ irods::check_directories_for_loops( boost::filesystem::path const & dirpath,
     try {
         // Default constructor creates an end iterator
         fs::recursive_directory_iterator end_itr;
-        const fs::symlink_option opt = (dashdashlink? fs::symlink_option::none : fs::symlink_option::recurse);
+        const fs::directory_options opt =
+            (dashdashlink ? fs::directory_options::none : fs::directory_options::follow_directory_symlink);
```

Comment:
Given that many of the distros we support have boost 1.7X in their repos, let's document somewhere that we made this change for boost 1.81 compatibility when upgrading from 1.78. This information may be useful once we start relying on distro-provided boost.

Reply:
Good idea.

Reply:
Capture in issue #7282.